pkg/watcher/results/results_test.go

```diff
@@ -0,0 +1,215 @@
+package results
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/results/pkg/watcher/convert"
+	"github.com/tektoncd/results/pkg/watcher/internal/test"
+	pb "github.com/tektoncd/results/proto/v1alpha2/results_go_proto"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/testing/protocmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestDefaultName(t *testing.T) {
+	// No need to create a full client - we're only testing a utility method.
+	client := &Client{kind: "test"}
+	want := "test-name"
+
+	objs := []metav1.Object{
+		&v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: "test",
+			},
+		},
+		&v1beta1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: "test",
+			},
+		},
+	}
+	for _, o := range objs {
+		t.Run(fmt.Sprintf("%T", o), func(t *testing.T) {
+			if got := client.defaultName(o); want != got {
+				t.Errorf("want %s, got %s", want, got)
+			}
+		})
+	}
+}
+
+func TestEnsureResult(t *testing.T) {
+	ctx := context.Background()
+	client := client(t)
+
+	objs := []metav1.Object{
+		&v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "taskrun",
+				Namespace: "test",
+			},
+		},
+		&v1beta1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
```

Comment:
Do we have a test somewhere that includes a Task/PipelineRun with a Spec/Status as well? (prob. makes more sense as an e2e test.)

Reply:
Added this to the TaskRun reconciler test in pkg/watcher/reconciler/taskrun/taskrun_test.go.